src/script/descriptor.cpp

```diff
@@ -756,6 +782,17 @@ class PKHDescriptor final : public DescriptorImpl
     PKHDescriptor(std::unique_ptr<PubkeyProvider> prov) : DescriptorImpl(Vector(std::move(prov)), "pkh") {}
     std::optional<OutputType> GetOutputType() const override { return OutputType::LEGACY; }
     bool IsSingleType() const final { return true; }
+
+    std::optional<int64_t> ScriptSize() const override { return 1 + 1 + 1 + 20 + 1 + 1; }
+
+    std::optional<int64_t> MaxSatSize(bool use_max_sig) const override {
+        const auto sig_size = use_max_sig ? 72 : 71;
+        return 1 + sig_size + 1 + 33;
```

Comment:
In c8b0aeb534edce08366176275d5577c1e309eb6f "descriptor: introduce a method to get the satisfaction size"

We could have an uncompressed pubkey, so we should get the size from the `PubkeyProvider`.

Reply:
Ah good catch! Done, thanks.